ensemble-view/src/main/java/hec/ensembleview/mappings/StatisticsMap.java

```diff
@@ -19,11 +19,16 @@ public void addStatisticsMapChangeListener(PropertyChangeListener listener) {
     }
 
     public void setTimeStatisticsMap() {
-        this.timeSeriesMapList = getTimeSeriesMapList();
+        if(this.timeSeriesMapList.size() == 0) {
+            return;
+        }
         support.firePropertyChange("time", false, true);
     }
 
     public void setEnsembleStatisticsMap() {
+        if(this.ensembleSeriesMapList.size() == 0) {
```

Comment:
Instead of checking the list size and seeing if it is equal to zero, use of the `isEmpty` method is generally preferred. I.e. `if (this.timeSeriesMapList.size().isEmpty) {}`

Reply:
thanks, made the update